proposals/lip-0057.md

```diff
@@ -1298,6 +1453,20 @@ def getDelegate(address: Address)-> DelegateStoreObject:
     return delegateStore(address)
 ```
 
+#### unbanDelegate
+
+Removes the banning for a delegate, given that the reason of banning was not due to misbehaviors. For example, this applies in cases the delegate got banned due to inactivity or not registering valid BLS keys.
+
+```python
+def unbanDelegate(address: Address)-> None:
+    if there does not exist an entry in delegate substore with key == address:
+        raise Exception('No registered delegate with the specified address.')
+    if  delegateStore(address).isBanned == False:
+        raise Exception('The specified delegate is not banned.')
+    if len(delegateStore(address).pomHeights) < POM_LIMIT_BANNED and Validators.getValidatorAccount(address).blsKey != INVALID_BLS_KEY:
```

Comment:
What is the reason for checking `Validators.getValidatorAccount(address).blsKey != INVALID_BLS_KEY` here? Is the reasoning that you should only be able to unban if you previously updated the BLS keys?

In a sense, what we want is also that you can only unban yourself if beforehand your BLS key was `INVALID_BLS_KEY`. This is not checked here, but in the legacy module. 

I would maybe try to avoid this coupling and have rather a simple, clean unbanning API that assumes it is used responsibly and where the conditions are checked outside the scope of this function. In particular, no knowledge of the regenesis is then required here.

Reply:
The checks included here were the ones that somehow are not necessarily performed before for registered delegates(either on delegate registration or genesis data). 

There is no coupling, exactly because we don't check if before the key was valid or not. Therefore, I don't see any benefit on removing this check to be honest, but could do it if you insist ;-)

Reply:
Ok I  see now: in your example, we should allow unbanning a delegate which is already banned from mainchain, even if they don't have BLS keys. Makes sense.  Removed the check.

Reply:
Looks good.